react/src/filter/Filters.tsx

```diff
@@ -4,6 +4,10 @@ import MultiSelectInput from '../components/MultiSelectInput';
 import useCanvasState from '../state/useCanvasState';
 import { GetCanvasEmbedResponse } from '@/src/rust_types/GetCanvasEmbedResponse';
 
+// Cache mapping filterId to its filterOptions
+// We are exclusively using the initial filter options from the initial API response
+const filterOptionsCache = new Map<string, string[][]>();
```

Comment:
Why is global instead of React state? This feels like a foot gun; for example if someone ever has two canvases embedded in their app this will clash

Reply:
If `filterId` isn't unique across Canvases then you're right. Updated